contracts/Template.sol

```diff
@@ -1,106 +1,159 @@
-        (,base,) = repo.getLatest();
-
-        return base;
+    function newTokenAndInstance(
+        string _tokenName,
+        string _tokenSymbol,
+        string _id,
+        address[] _holders,
+        uint256[] _stakes,
+        uint64[3] _votingSettings /* supportRequired, minAcceptanceQuorum, voteDuration */
+    )
+        external
+    {
+        newToken(_tokenName, _tokenSymbol);
+        newInstance(_id, _holders, _stakes, _votingSettings);
     }
-}
-
-
-contract Template is TemplateBase {
-    MiniMeTokenFactory tokenFactory;
-
-    uint64 constant PCT = 10 ** 16;
-    address constant ANY_ENTITY = address(-1);
 
-    constructor(ENS ens) TemplateBase(DAOFactory(0), ens) public {
-        tokenFactory = new MiniMeTokenFactory();
+    function newToken(string _name, string _symbol) public returns (MiniMeToken) {
+        MiniMeToken token = _createToken(_name, _symbol, TOKEN_DECIMALS);
+        _cacheToken(token, msg.sender);
+        return token;
     }
 
-    function newInstance() public {
-        Kernel dao = fac.newDAO(this);
-        ACL acl = ACL(dao.acl());
-        acl.createPermission(this, dao, dao.APP_MANAGER_ROLE(), this);
-
-        address root = msg.sender;
-        bytes32 appId = keccak256(abi.encodePacked(apmNamehash("open"), keccak256("placeholder-app-name")));
-        bytes32 votingAppId = apmNamehash("voting");
-        bytes32 tokenManagerAppId = apmNamehash("token-manager");
-
-        CounterApp app = CounterApp(dao.newAppInstance(appId, latestVersionAppBase(appId)));
-        Voting voting = Voting(dao.newAppInstance(votingAppId, latestVersionAppBase(votingAppId)));
-        TokenManager tokenManager = TokenManager(dao.newAppInstance(tokenManagerAppId, latestVersionAppBase(tokenManagerAppId)));
-
-        MiniMeToken token = tokenFactory.createCloneToken(MiniMeToken(0), 0, "App token", 0, "APP", true);
-        token.changeController(tokenManager);
-
-        // Initialize apps
-        app.initialize();
-        tokenManager.initialize(token, true, 0);
-        voting.initialize(token, 50 * PCT, 20 * PCT, 1 days);
+    function newInstance(
+        string _id,
+        address[] _holders, 
+        uint256[] _stakes,
+        uint64[3] _votingSettings
+    ) 
+        public 
+    {
+        require(_holders.length > 0, ERROR_EMPTY_HOLDERS);
+        require(_holders.length == _stakes.length, ERROR_BAD_HOLDERS_STAKES_LEN);
+        require(_votingSettings.length == 3, ERROR_BAD_VOTE_SETTINGS);
+        MiniMeToken token = _popTokenCache(msg.sender);
+        // Create DAO and install apps
+        (Kernel dao, ACL acl) = _createDAO();
+        TokenManager tokenManager = _installTokenManagerApp(dao, token, TOKEN_TRANSFERABLE, TOKEN_MAX_PER_ACCOUNT);
+        Voting voting = _installVotingApp(dao, token, _votingSettings[0], _votingSettings[1], _votingSettings[2]);
+        // Install app
+        CounterApp app = _installCounterApp(dao);
```

Comment:
I would encapsulate all the "custom app" installation and permissions into one or two functions, called after we initialize the "base" apps, to make it super easy for people to extend.